services/workflows-service/src/main.ts

```diff
@@ -71,6 +75,8 @@ async function main() {
   app.useGlobalFilters(new AllExceptionsFilter(httpAdapter));
   app.useGlobalFilters(new HttpExceptionFilter(httpAdapter));
 
+  app.enableShutdownHooks();
```

Comment:
This line allows the usage of `onModuleDestroy` hook.
https://docs.nestjs.com/fundamentals/lifecycle-events#application-shutdown